File: task1/main.js
const readline = require('readline-sync');

const text = readline.question("Введите строку: ");
const glas = "aeiouyаеёиоуыэюяAEIOUYАЕЁИОУЫЭЮЯ";
const soglas = "bcdfghjklmnpqrstvwxzбвгджзйклмнпрстфхцчшщBCDFGHJKLMNPQRSTVWXZБВГДЖЗЙКЛМНПРСТФХЦЧШЩ";

let countGlas = 0, countSog = 0;

for (const ch of text) {
    if (glas.includes(ch)) countGlas++;
    else if (soglas.includes(ch)) countSog++;
}

if (countGlas > countSog) console.log("да");
else if (countGlas < countSog) console.log("нет");
else console.log("одинаково");

File: task1/main.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Введите строку: ");
        String text = scanner.nextLine(); // запись входной строки

        String glas = "aeiouyаеёиоуыэюяAEIOUYАЕЁИОУЫЭЮЯ";
        String soglas = "bcdfghjklmnpqrstvwxzбвгджзйклмнпрстфхцчшщBCDFGHJKLMNPQRSTVWXZБВГДЖЗЙКЛМНПРСТФХЦЧШЩ";

        int countGlas = 0, countSog = 0; // задаем счетчик для гласных и согласных
        
        // проходимся циклом по строке
        for (char ch : text.toCharArray()) {
            if (glas.indexOf(ch) != -1) {
                countGlas++;
            } else if (soglas.indexOf(ch) != -1) {
                countSog++;
            }
        }
        
        // делаем вывод по условию задачи
        if (countGlas > countSog) {
            System.out.println("да");
        } else if (countGlas < countSog) {
            System.out.println("нет");
        } else {
            System.out.println("одинаково");
        }
    }
}
